manager/state/raft/raft.go

```diff
@@ -1296,25 +1296,32 @@ func (n *Node) sendToMember(ctx context.Context, members map[uint64]*membership.
 	defer n.asyncTasks.Done()
 	defer close(thisSend)
 
-	ctx, cancel := context.WithTimeout(ctx, n.opts.SendTimeout)
-	defer cancel()
-
 	if lastSend != nil {
+		waitCtx, waitCancel := context.WithTimeout(ctx, n.opts.SendTimeout)
+		defer waitCancel()
+
 		select {
```

Comment:
> Rather than having two selects, just set `lastSend` to `nil`.

I'm not sure how to do that. One of the selects has a default case, and the other does not.

The intent is to make sure we aren't continuing if the context timeout has been reached. Suppose `lastSend` was closed due to timing out, and our context also timed out around the same time, but since `select` picks a random case of those which can proceed, we might end up in `<-lastSend` instead of `<-waitCtx.Done()`.